src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java

```diff
@@ -1,24 +1,620 @@
+
+  /** The name of the IO server for the tag that is used to halt the connector execution. */
+  public static final String CONNECTOR_HALT_TAG_IO_SERVER_NAME = "MEM";
+
+  /** The type of the tag (boolean) that is used to halt the connector execution. */
+  public static final int CONNECTOR_HALT_TAG_TYPE = 0;
+
+  /** The description of the tag that is used to halt the connector execution. */
+  public static final String CONNECTOR_HALT_TAG_DESCRIPTION =
+      "Tag which is used to halt the execution of the Cumulocity connector application.";
+
+  /** The value of the connector halt tag that permits the application to execute. */
+  public static final int CONNECTOR_CONTROL_TAG_RUN_VALUE = 0;
+
+  /** The interval (in milliseconds) at which the main loop is executed. */
+  private static final int MAIN_LOOP_CYCLE_TIME_MILLIS = 1000;
+
+  /**
+   * The HTTP connection timeout time in seconds. This value affects the Ewon's global HTTP
+   * timeouts.
+   */
+  public static final String HTTP_TIMEOUT_SECONDS_STRING = "10";
+
+  /**
+   * The prefix appended to the front of the Ewon Flexy's serial number to form the MQTT client ID.
+   */
+  private static final String MQTT_CLIENT_ID_SERIAL_PREFIX = "HMS-Flexy-";
+
+  /** MQTT manager */
+  private static CConnectorMqttMgr mqttMgr;
+
+  /** Alarm manager */
+  private static CConnectorAlarmMgr alarmMgr;
+
+  /** Connector configuration object */
+  private static CConnectorConfigFile connectorConfig;
+
+  /** Connector control tag object for accessing its value. */
+  private static TagControl connectorControlTag = null;
+
+  /** Boolean flag to indicate if the connector is running. */
+  private static boolean isRunning = true;
+
+  /**
+   * Serial number of the host Ewon device, populated during the initialization of the connector.
+   */
+  private static String ewonSerialNumber = "";
+
+  /** Boolean indicating if the device should be restarted after the connector shuts down. */
+  private static boolean restartDeviceAfterShutdown = false;
+
+  /**
+   * Gets the connector MQTT manager.
+   *
+   * @return connector MQTT manager
+   */
+  public static CConnectorMqttMgr getMqttMgr() {
+    return mqttMgr;
+  }
+
+  /**
+   * Gets the connector configuration object.
+   *
+   * @return connector configuration
+   */
+  public static CConnectorConfigFile getConnectorConfig() {
+    return connectorConfig;
+  }
+
+  /** Triggers a shutdown of the connector and restart of the host device when requested. */
+  public static void shutdownConnectorAndRestartDevice() {
+    Logger.LOG_CRITICAL("The connector has been requested to shut down and restart the device...");
+    restartDeviceAfterShutdown = true;
+    isRunning = false;
+  }
+
+  /** Method for performing connector application initialization steps. */
+  private static boolean initialize() {
+    Logger.LOG_CRITICAL("Initializing " + CONNECTOR_FRIENDLY_NAME + "...");
```

Comment:
please add   SCAppManagement.enableAppAutoRestart();

Reply:
Functionality changes are being added to dev/1.0.0. In the interest of time, I wanted to get this PR through review.

Reply:
See: https://github.com/hms-networks/flexy-cumulocity-connector/commit/874214802c13dc7d0e89a40103d3603df01f8a18